cookbooks/aws-parallelcluster-config/files/default/cloudwatch_agent/write_cloudwatch_agent_json.py

```diff
@@ -177,14 +178,68 @@ def filter_output_fields(configs):
     return [{desired_key: config[desired_key] for desired_key in desired_keys} for config in configs]
 
 
-def create_config(log_configs):
+def create_metrics_collected(selected_configs):
+    """Create the "metrics_collected" section in metrics configuration for selected metrics."""
+    metrics_collected = {}
+    for metric_config in selected_configs:
+        metric_key = metric_config["metric_type"]
+        metrics_collected[metric_key] = {
+            "measurement": metric_config["measurement"],
+            "metrics_collection_interval": metric_config.get(
+                "metrics_collection_interval", DEFAULT_METRICS_COLLECTION_INTERVAL
+            ),
+        }
+        if "resources" in metric_config.keys():
+            metrics_collected[metric_key]["resources"] = metric_config["resources"]
+        if "append_dimensions" in metric_config.keys() and "ClusterName" in metric_config["append_dimensions"]:
+            metrics_collected[metric_key]["append_dimensions"] = {"ClusterName": get_node_info().get("stack_name")}
+    return metrics_collected
+
+
+def select_metrics(configs, args):
+    """Add metrics_collected for CloudWatch Agent Metrics section."""
+    selected_configs = select_configs_for_node_role(configs["metrics_collected"], args.node_role)
+    metric_configs = {"metrics_collected": create_metrics_collected(selected_configs)}
+    return metric_configs
+
+
+def select_append_dimensions(dimensions):
+    """Create the dictionary of append dimensions according to a list of dimension names."""
+    valid_append_dimensions = {
+        "ImageID": "${aws:ImageId}",
+        "InstanceId": "${aws:InstanceId}",
+        "InstanceType": "${aws:InstanceType}",
+        "AutoScalingGroupName": "${aws:AutoScalingGroupName}",
+    }
+    append_dimensions = {key: valid_append_dimensions[key] for key in dimensions}
+    return append_dimensions
+
+
+def add_append_dimensions(metric_configs, configs):
+    """Add the "append_dimensions" filed for the CloudWatch Agent Metrics section."""
+    if "append_dimensions" in configs.keys() and len(metric_configs["metrics_collected"]) > 0:
```

Comment:
nit: we can remove `.keys()` here

Reply:
Also too, I think you don't need `and len(metric_configs["metrics_collected"]) > 0:` as just `and metric_configs["metrics_collected"]:` should work I think?